### ISSUE TITLE
Feature Request: Add UPNP Support.

### ISSUE THREAD

**Issue by ConnorBP** [NiceNode/nice-node#430]
## Problem details

Some home internet routers these days purposefully block users from using manual port forwarding. They do however still support UPNP for gaming services and such. Having a gui node like this, the use case seems to be mostly for home users. This being the case, I think it would be a very valuable feature for many home users to get up and running with no hassle.

In my case, unfortunately, I need to buy a new router and put my ISP router in passthrough mode (which is expensive). Otherwise I can switch ISPs (might do) or somehow find a way to manually push a UPNP.

What went wrong and what should have happened?

I started the program, it said my ports are closed. I checked my windows firewall to make sure it was letting things through, then checked my router and added the ports to its port forwarding settings. Then when it was not working I found out that my ISP modified the firmware to ignore those and only allow UPNP.

What should have happened:

Program starts, it handles the ports behind the scenes, and then the user does not need to worry about it.

To summarize, UPNP would be a very good convenience feature for many users, and it is currently necessary for me.

If there is a better place to submit feature requests than here, please let me know :) Thank you.

**Comment by jgresham** [comment]
Hey, thanks for the feature request. I'll check this out.

**Comment by ConnorBP** [comment]
Update: found out that windows firewall still blocks all ports when the firewall is set to "disabled". I added the forward rule and it works fine now. I would love if nice-node supported both UPNP as well as automatic windows firewall rule addition. These two features together would make setup truly 1-click for a home windows PC user.

**Comment by jgresham** [comment]
> Update: found out that windows firewall still blocks all ports when the firewall is set to "disabled". I added the forward rule and it works fine now. I would love if nice-node supported both UPNP as well as automatic windows firewall rule addition. These two features together would make setup truly 1-click for a home windows PC user.

absolutely

**Comment by jgresham** [comment]
+1 If you have any links or commands on hand to help us add this more quickly
